massa-api/src/public.rs

```diff
@@ -499,24 +480,75 @@ impl Endpoints for API<Public> {
                         .chain(get_consensus_ops?.into_keys())
                         .collect();
 
-                        let get_pool_eds = pool_cmd_snd.get_endorsements_by_address(address);
-                        let get_consensus_eds = cmd_snd.get_endorsements_by_address(address);
-                        let (get_pool_eds, get_consensus_eds) =
-                            tokio::join!(get_pool_eds, get_consensus_eds);
-                        let gathered_ed: Set<EndorsementId> = get_pool_eds?
-                            .into_keys()
-                            .chain(get_consensus_eds?.into_keys())
-                            .collect();
-                    Result::<(Address, Set<BlockId>, Set<OperationId>, Set<EndorsementId>), ApiError>::Ok((
-                        address, blocks, gathered, gathered_ed
+                    let get_pool_eds = pool_cmd_snd.get_endorsements_by_address(address);
+                    let get_consensus_eds = cmd_snd.get_endorsements_by_address(address);
+                    let (get_pool_eds, get_consensus_eds) =
+                        tokio::join!(get_pool_eds, get_consensus_eds);
+                    let gathered_ed: Set<EndorsementId> = get_pool_eds?
+                        .into_keys()
+                        .chain(get_consensus_eds?.into_keys())
+                        .collect();
+
+                    let (final_sce, candidate_sce) =
+                        match exec_snd.get_final_and_active_ledger_entry(&address) {
+                            (None, None) => (SCELedgerInfo::default(), SCELedgerInfo::default()),
+                            (None, Some(candidate)) => (
+                                SCELedgerInfo::default(),
+                                SCELedgerInfo {
+                                    balance: candidate.parallel_balance,
+                                    module: candidate.bytecode,
+                                    datastore: candidate.datastore.into_iter().collect(),
```

Comment:
Why not having the same type in ledger info and ledger entry ? For example having a BTreeMap on LedgerInfo to avoid this conversion very costly. (except if rust optimize it)

Reply:
I don't know, maybe it's in the api specs @damip @yvan-sraka

Reply:
I don't know either ...

Reply:
#2397